internal/jobservice/repository/sql_job_service.go

```diff
@@ -0,0 +1,220 @@
+	_, errIndex := s.db.Exec(`CREATE INDEX idx_job_set_queue 
+ON jobservice (Queue, JobSetId)`)
+	if errIndex != nil {
+		panic(errIndex)
+	}
+}
+
+// Get the JobStatus given the jodId
+func (s *SQLJobService) GetJobStatus(jobId string) (*js.JobServiceResponse, error) {
+	row := s.db.QueryRow("SELECT JobResponseState, JobResponseError FROM jobservice WHERE JobId=?", jobId)
+	var jobState string
+	var jobError string
+	err := row.Scan(&jobState, &jobError)
+	if err == sql.ErrNoRows {
+		return &js.JobServiceResponse{State: js.JobServiceResponse_JOB_ID_NOT_FOUND}, nil
+	} else if err != nil {
+		return nil, err
+	}
+	jobProtoResponse := &js.JobServiceResponse{Error: jobError}
+	switch jobState {
+	case "SUBMITTED":
+		jobProtoResponse.State = js.JobServiceResponse_SUBMITTED
+	case "DUPLICATE_FOUND":
+		jobProtoResponse.State = js.JobServiceResponse_DUPLICATE_FOUND
+	case "RUNNING":
+		jobProtoResponse.State = js.JobServiceResponse_RUNNING
+	case "FAILED":
+		jobProtoResponse.State = js.JobServiceResponse_FAILED
+	case "SUCCEEDED":
+		jobProtoResponse.State = js.JobServiceResponse_SUCCEEDED
+	case "CANCELLED":
+		jobProtoResponse.State = js.JobServiceResponse_CANCELLED
+	case "JOB_ID_NOT_FOUND":
+		jobProtoResponse.State = js.JobServiceResponse_JOB_ID_NOT_FOUND
+	}
+	return jobProtoResponse, nil
+}
+
+// Update in memory JobStatus Map with jobId and our JobTable
+func (s *SQLJobService) UpdateJobServiceDb(jobTable *JobTable) {
+	stmt, err := s.db.Prepare("INSERT OR REPLACE INTO jobservice VALUES (?, ?, ?, ?, ?, ?)")
+	if err != nil {
+		panic(err)
+	}
+	jobState := jobTable.jobResponse.State.String()
+	_, errExec := stmt.Exec(jobTable.queue, jobTable.jobSetId, jobTable.jobId, jobState, jobTable.jobResponse.Error, jobTable.timeStamp)
+
+	if errExec != nil {
+		panic(errExec)
+	}
+}
+
+// Simple Health Check to Verify if SqlLite is working.
+func (s *SQLJobService) HealthCheck() (bool, error) {
+	row := s.db.QueryRow("SELECT 1")
+	var col int
+	err := row.Scan(&col)
+	if err == nil {
+		return true, nil
+	} else {
+		return false, fmt.Errorf("SQL health check failed: %v", err)
+	}
+}
+
+// Check if JobSet is in our map.
+// Note: The key should be queuejobset.
+func (s *SQLJobService) IsJobSetSubscribed(queue string, jobSet string) bool {
+	s.jobSetSubscribe.subscribeLock.Lock()
+	defer s.jobSetSubscribe.subscribeLock.Unlock()
+	primaryKey := queue + jobSet
+	_, ok := s.jobSetSubscribe.subscribeMap[primaryKey]
+	return ok
+}
+
+// Mark our JobSet as being subscribed
+// SubscribeTable contains Queue, JobSet and time when it was created.
+func (s *SQLJobService) SubscribeJobSet(queue string, jobSet string) {
+	s.jobSetSubscribe.subscribeLock.Lock()
+	defer s.jobSetSubscribe.subscribeLock.Unlock()
+	primaryKey := queue + jobSet
+	_, ok := s.jobSetSubscribe.subscribeMap[primaryKey]
+	if !ok {
+		s.jobSetSubscribe.subscribeMap[primaryKey] = NewSubscribeTable(queue, jobSet)
+	}
+
+}
+
+// UnSubscribe to JobSet and delete all the jobs in the in memory map
```

Comment:
thinkt he comment might be out of date as now it deletes the jobs in the db?

Reply:
Also technically this function doesn't unsubscribe (i.e. stop the goroutine with the subscription).  I think it should be renamed `cleanupUnsubscribedData` or something